src/hotspot/os/bsd/memMapPrinter_macosx.cpp

```diff
@@ -0,0 +1,375 @@
+      st->print(" ");
+    } else {
+      const char* tag = mapping_info._tag_text;
+      if (tag != NULL) {
+        st->print("[%s] ", tag);
+      }
+    }
+    st->print_raw(mapping_info._file_name.base());
+  #undef INDENT_BY
+    st->cr();
+  }
+
+  void print_legend() const {
+    outputStream* st = _session.out();
+    st->print_cr("from, to, vsize: address range and size");
+    st->print_cr("prot:    protection:");
+    st->print_cr("           rwx: read / write / execute");
+    st->print_cr("share:   share mode:");
+    st->print_cr("           cow: copy on write");
+    st->print_cr("           pvt: private");
+    st->print_cr("           shr: shared");
+    st->print_cr("           tsh: true shared");
+    st->print_cr("           p/a: private aliased");
+    st->print_cr("           s/a: shared aliased");
+    st->print_cr("           lpg: large page");
+    st->print_cr("offset:  offset from start of allocation block");
+    st->print_cr("vminfo:  VM information (requires NMT)");
+    st->print_cr("file:    file mapped, if mapping is not anonymous");
+    {
+      streamIndentor si(st, 16);
+      _session.print_nmt_flag_legend();
+    }
+    st->print_cr("file:            file mapped, if mapping is not anonymous");
+  }
+
+  void print_header() const {
+    outputStream* st = _session.out();
+    //            0         1         2         3         4         5         6         7         8         9         0         1         2         3         4         5         6         7
+    //            012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789
+    //            0x00000001714a0000-0x000000017169c000      2080768 rw-/rwx  p/a       0xc000 STACK-28419-C2-CompilerThread0
+    st->print_cr("from               to                        vsize prot share     offset vminfo/file");
+    st->print_cr("==================================================================================================");
+  }
+};
+
+void MemMapPrinter::pd_print_all_mappings(const MappingPrintSession& session) {
+
+  ProcSmapsPrinter printer(session);
+  ProcSmapsSummary summary;
+  outputStream* const st = session.out();
+  const pid_t pid = getpid();
+
+  printer.print_legend();
+  st->cr();
+  printer.print_header();
+
+  proc_regionwithpathinfo region_info;
+  MappingInfo mapping_info;
+  uint64_t address = 0;
+  int region_count = 0;
+  while (true) {
+    if (++region_count > MAX_REGIONS_RETURNED) {
+      st->print_cr("limit of %d regions reached (results inaccurate)", region_count);
+      break;
+    }
+    int retval = proc_pidinfo(pid, PROC_PIDREGIONPATHINFO, (uint64_t)address, &region_info, sizeof(region_info));
+    if (retval <= 0) {
+      break;
+    } else if (retval < (int)sizeof(region_info)) {
+      fatal("proc_pidinfo() returned %d", retval);
```

Comment:
I would make this an assert, and return without printing anything in release.

Reply:
I missed this.  I have changed it to the Windows version, where it prints the error and has an assert(), instead of fatal().